### PR TITLE
Support replication mode control

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

Part of #7097.

This PR adds a global group registry that stores the group ID of every met store. The information will be used for further implement of label based replication.

### Check List 

Tests 

- Unit test
- Integration test

### CODE REVIEW

File: components/test_raftstore/src/pd.rs [hunk 1]
```diff
@@ -311,6 +315,20 @@ impl Cluster {
         }
     }
 
+    fn get_all_stores(&self) -> Result<Vec<metapb::Store>> {
+        Ok(self
```
Comment (hicqu) [inline components/test_raftstore/src/pd.rs:319]: Why filter stores with id == 0? Is it possible?

Comment (BusyJay) [reply components/test_raftstore/src/pd.rs:319]: Yes. It's zero means it's a placeholder during testing cluster initializing:

https://github.com/tikv/tikv/blob/90d0739beb6b2213bb2e3db314226b080995e019/components/test_raftstore/src/pd.rs#L288-L305

**Comment by hicqu** [comment]
LGTM.

File: components/raftstore/src/store/replication_mode.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    self.labels.get(&label),
+                    Some(o.get()),
+                    "store {:?}",
+                    store_id
+                );
+                *o.get()
+            }
+            HashMapEntry::Vacant(v) => {
+                let len = self.labels.len() as u64 + 1;
+                let id = *self.labels.entry(label).or_insert(len);
```
Comment (gengliqi) [inline components/raftstore/src/store/replication_mode.rs:37]: Why the length of label can be defined as group id?

Comment (BusyJay) [reply components/raftstore/src/store/replication_mode.rs:37]: Group ID is just an arbitrary number that maps to label value.

**Pull request merged**